index.js

```diff
@@ -1,9 +1,32 @@
 'use strict';
+const URL = require('url').Url;
 const urlRegex = require('url-regex');
 const normalizeUrl = require('normalize-url');
+const unescape = require('unescape');
 
 module.exports = (str, opts) => {
-	const urls = str.match(urlRegex());
-	const ret = urls ? urls.map(url => normalizeUrl(url.trim().replace(/\.+$/, ''), opts)) : [];
-	return new Set(ret);
+	opts = opts || {};
+	const ret = new Set();
+
+	function _extractQueryParams(url) {
```

Comment:
This should be a top-level function that *just* extracts the URLs from query params and returns the URLs.

Reply:
done